src/BenchmarkDotNet/Attributes/Filters/OperatingSystemsFilterAttribute.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Linq;
+using BenchmarkDotNet.Filters;
+using JetBrains.Annotations;
+using System.Runtime.InteropServices;
+
+namespace BenchmarkDotNet.Attributes
+{
+    [PublicAPI]
+    public class OperatingSystemsFilterAttribute : FilterConfigBaseAttribute
+    {
+        // CLS-Compliant Code requires a constructor without an array in the argument list
+        public OperatingSystemsFilterAttribute() { }
+
+        /// <param name="allowed">if set to true, the OSes beloning to platforms are enabled, if set to false, disabled</param>
+        public OperatingSystemsFilterAttribute(bool allowed, params PlatformID[] platforms)
+            : base(new SimpleFilter(_ =>
+            {
+                return allowed
+                    ? platforms.Any(platform => RuntimeInformation.IsOSPlatform(Map(platform)))
```

Comment:
Maybe you could take the `string` of a OSPlatform instead of `PlatformID`? (basically the string that gets passed into https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.osplatform.create)

I don't think PlatformID is going to work.

From https://docs.microsoft.com/en-us/dotnet/api/system.platformid

> MacOSX | 6 | The operating system is Macintosh. This value was returned by Silverlight. On .NET Core, its replacement is Unix.

From that, it sounds like you won't be able to distinguish between macOS and Linux on .NET Core using PlatformID.

Reply:
@eerhardt I am not sure about string. For example: let's say that I dont't want to run the benchmark on Ubuntu. Should I put `Unix`? `Linux`? or `Ubuntu`? or `Ubuntu16.04`?

But I agree that PlatformID is not perfect, I had to implement a mapping on my own.

Maybe I should just introduce a new enum?

Reply:
> Should I put Unix? Linux? or Ubuntu? or Ubuntu16.04?

Do you want that deep of control? Do you want to be able to say "All Unixes" vs. "just Ubuntu, but not RedHat or OpenSUSE"? If so, it seems like a perfect place to use RIDs...

Reply:
> Do you want that deep of control?

No, I just wanted to show that without an enum, the user needs to guess the right string value.

Reply:
It's been two years since I created this PR, but now I really need tha for Randomization in the perf repo so I've updated it. I decided to include my own simple enum, just so users don't have to type OS names manually (an avoid mistakes for "OSX" != "macOS" etc)